From 8437f4f30ad89f996831ce91a901682bc4184d07 Mon Sep 17 00:00:00 2001
From: Liu Ying <Ying.Liu@freescale.com>
Date: Thu, 26 Nov 2015 16:11:17 +0800
Subject: [PATCH] MLK-11911-3 mxc IPUv3: disp: Correct display ID check in
 ipu_uninit_sync_panel()

This patch fixes the following issue reported by Coverity:
Constant expression result (CONSTANT_EXPRESSION_RESULT)
always_true_or: The "or" condition disp != 0 || disp != 1 will always be true
because disp cannot be equal to two different values at the same time, so it
must be not equal to at least one of them.
	if ((disp != 0) || (disp != 1))
		return;

Signed-off-by: Liu Ying <Ying.Liu@freescale.com>
(cherry picked from commit c700631cee685a4c426380e14e50b04e4d635dbf)
---
 drivers/mxc/ipu3/ipu_disp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mxc/ipu3/ipu_disp.c b/drivers/mxc/ipu3/ipu_disp.c
index f7044c2..db182ed 100644
--- a/drivers/mxc/ipu3/ipu_disp.c
+++ b/drivers/mxc/ipu3/ipu_disp.c
@@ -1561,7 +1561,7 @@ void ipu_uninit_sync_panel(struct ipu_soc *ipu, int disp)
 	uint32_t reg;
 	uint32_t di_gen;
 
-	if ((disp != 0) || (disp != 1))
+	if (disp != 0 && disp != 1)
 		return;
 
 	mutex_lock(&ipu->mutex_lock);
-- 
1.8.0

